core/src/commonMain/kotlin/org/jetbrains/litmuskt/LitmusRunner.kt

```diff
@@ -1,65 +1,23 @@
 package org.jetbrains.litmuskt
 
 import kotlin.time.Duration
-import kotlin.time.TimeSource
 
 abstract class LitmusRunner {
 
     /**
      * Starts threads for the test and returns a "join handle". This handle should block
      * until the threads join and then collect and return the results.
+     *
+     * It should also reset() the states after running the test.
      */
-    protected abstract fun <S : Any> startTest(
+    abstract fun <S : Any> startTest(
         test: LitmusTest<S>,
         states: Array<S>,
         barrierProducer: BarrierProducer,
         syncPeriod: Int,
         affinityMap: AffinityMap?,
     ): BlockingFuture<LitmusResult>
 
-    /**
-     * Entry point for running tests. This method can be overridden in case that particular runner
-     * does not need to allocate states.
-     */
-    open fun <S : Any> startTest(test: LitmusTest<S>, params: LitmusRunParams): BlockingFuture<LitmusResult> {
-        val states = TypedArray(params.batchSize) { test.stateProducer() }
-        return startTest(test, states, params.barrierProducer, params.syncPeriod, params.affinityMap)
-    }
```

Comment:
It would be good to keep it as `protected` inside the interface, but now that the scary `Any` type is gone from its signature, I agree it is better to move it out.